turbopack/crates/turbo-tasks/src/manager.rs

```diff
@@ -627,28 +671,32 @@ impl<B: Backend + 'static> TurboTasks<B> {
         if registry::get_function(func).arg_meta.is_resolved(&*arg) {
             return self.native_call(func, arg, persistence);
         }
+        let task_type = CachedTaskType::ResolveNative {
+            fn_type: func,
+            this: None,
+            arg,
+        };
+        #[cfg(feature = "local_resolution")]
+        return CURRENT_GLOBAL_TASK_STATE.with(move |gts| {
+            let mut gts_write = gts.write().unwrap();
+            let local_task_id = gts_write.create_local_task(task_type, persistence);
+            RawVc::LocalOutput(gts_write.task_id, local_task_id)
+        });
+        #[cfg(not(feature = "local_resolution"))]
         match persistence {
             TaskPersistence::LocalCells => {
```

Comment:
Looks like `TaskPersistence::LocalCells` is not needed for local tasks at all. Maybe we could remove this variant (resp. modify the previous PR to never introduce it).

I like that local tasks are completely handled by the manager and not leaking into the backend.